vulcanai2/tests/models/test_cnn.py

```diff
@@ -131,27 +144,64 @@ def test_unfreeze_noclass(self, conv3D_net):
             assert params.requires_grad is True
 
     def test_fit_multi_input(self, multi_input_cnn,
-                             multi_input_cnn_data):
-        """Test for fit function"""
-        test_train = Subset(multi_input_cnn_data, 
-                            range(len(multi_input_cnn_data)//2))
-        test_val = Subset(multi_input_cnn_data, 
-                          range(len(multi_input_cnn_data)//2, 
-                                    len(multi_input_cnn_data)))
-        test_train_loader = DataLoader(test_train, batch_size=2)
-        test_val_loader = DataLoader(test_val, batch_size=2)     
-        
-        init_weights = copy.deepcopy(multi_input_cnn.network[0]._kernel.weight.data)
+                             multi_input_train_loader,
+                             multi_input_test_loader):
+        """Test for fit function"""        
+        init_weights = copy.deepcopy(multi_input_cnn.network[0]._kernel.weight.detach())
         multi_input_cnn_no_fit = copy.deepcopy(multi_input_cnn)
         try:
-            multi_input_cnn.fit(test_train_loader, test_val_loader, 2)
+            multi_input_cnn.fit(multi_input_train_loader, 
+                                multi_input_test_loader, 2)
         except RuntimeError:
+            #TODO: Change to logger
             print("The network multi_input_cnn failed to train.")
         finally:
-            trained_weights = multi_input_cnn.network[0]._kernel.weight.data.cpu()
+            trained_weights = multi_input_cnn.network[0]._kernel.weight.detach().cpu()
             
             # Sanity check if the network parameters are training
             assert (torch.equal(init_weights.cpu(), trained_weights.cpu()) is False)
             assert (torch.equal(list(multi_input_cnn.state_dict().values())[0],
                                 list(multi_input_cnn_no_fit.state_dict().values())[0])\
-                                is False)
\ No newline at end of file
+                                is False)
+    
+    def test_params_multi_input(self, multi_input_cnn,
+                                multi_input_train_loader,
+                                multi_input_test_loader):
+        """Test for change in network params/specifications"""
+        
+        test_net1 = copy.deepcopy(multi_input_cnn)
+        test_net2 = pickle.loads(pickle.dumps(multi_input_cnn))
```

Comment:
I'm not sure why you are doing this. Would you be able to clarify?

Reply:
just wanted to make sure if copying/duplicating the models via `deepcopy` or `pickle` does copy all the parameters? If you ask about a use-case for copying: I thought about what if we have a certain network module within a multi-input network, make a copy of it and insert it somewhere else within the same multi-input network using `add_input_network`. Not sure how feasible that is.

Reply:
sure. you can do that as long as the dimensions of the inputs and outputs agree. it just becomes another independent module if you `deepcopy` it.

Reply:
I can remove `deepcopy` testing in that case. Just wanted to prove the users that it is possible to use both copying techniques.

Reply:
might take up a lot of memory to copy several times

Reply:
removed it now